### PR TITLE
Expose running and queued call counts for monitoring

### PR DESCRIPTION
**Pull request by kryali**
goal: add monitoring around how many requests are queued, in-flight, and maximum capacity.

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java [hunk 1]
```diff
@@ -177,4 +177,12 @@ synchronized void executed(Call call) {
   synchronized void finished(Call call) {
     if (!executedCalls.remove(call)) throw new AssertionError("Call wasn't in-flight!");
   }
+
+  public int getRunningCallCount() {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java]: Needs `synchronized`.

File: okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java [hunk 2]
```diff
@@ -177,4 +177,12 @@ synchronized void executed(Call call) {
...
+    return runningCalls.size();
+  }
+
+  public int getQueuedCallCount() {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java]: Needs `synchronized`.

**Comment by swankjesse** [comment]
Nice.

File: okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java (hunk 2 above)
Comment (kryali) [inline okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java]: Added.

Also, I have an irrational fear that metrics systems may call this too many times and then interfere with actual request processing. That's irrational right?

Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Dispatcher.java]: Not irrationally, but unlikely. If it comes up, we'll figure something out.

**Pull request merged**